src/controller/python/chip/clusters/Attribute.py

```diff
@@ -65,17 +78,92 @@ def __hash__(self):
         return str(self).__hash__()
 
 
+@dataclass
+class EventPath:
+    EndpointId: int = None
+    ClusterId: int = None
+    EventId: int = None
+
+    def __init__(self, EndpointId: int = None, Cluster=None, Event=None, ClusterId=None, EventId=None):
+        self.EndpointId = EndpointId
+        if Cluster is not None:
+            # Wildcard read for a specific cluster
+            if (Event is not None) or (ClusterId is not None) or (EventId is not None):
+                raise Warning(
+                    "Event, ClusterId and AttributeId is ignored when Cluster is specified")
+            self.EventId = Event.id
+            return
+        if Event is not None:
+            if (ClusterId is not None) or (EventId is not None):
+                raise Warning(
+                    "ClusterId and EventId is ignored when Event is specified")
+            self.ClusterId = Event.cluster_id
+            self.EventId = Event.event_id
+            return
+        self.ClusterId = ClusterId
+        self.EventId = EventId
+
+    def __str__(self) -> str:
+        return f"{self.EndpointId}/{self.EventId}/{self.EventId}"
+
+    def __hash__(self):
+        return str(self).__hash__()
+
+
 @dataclass
 class AttributePathWithListIndex(AttributePath):
     ListIndex: int = None
 
 
+@dataclass
+class EventHeader:
+    EndpointId: int = None
+    ClusterId: int = None
+    EventId: int = None
+    EventNumber: int = None
+    Priority: EventPriority = None
+    Timestamp: int = None
+    TimestampType: EventTimestampType = None
+
+    def __init__(self, EndpointId: int = None, Cluster=None, Event=None, ClusterId=None, EventId=None, EventNumber=None, Priority=None, Timestamp=None, TimestampType=None):
+        self.EndpointId = EndpointId
+        if Cluster is not None:
+            # Wildcard read for a specific cluster
+            if (Event is not None) or (ClusterId is not None) or (EventId is not None):
+                raise Warning(
+                    "EventId, ClusterId and EventId is ignored when Cluster is specified")
+            self.ClusterId = Cluster.id
+            return
+        if Event is not None:
+            if (ClusterId is not None) or (EventId is not None):
```

Comment:
ditto: line 89
```suggestion
        if Cluster:
            # Wildcard read for a specific cluster
            if Event or ClusterId or EventId:
                raise Warning(
                    "EventId, ClusterId and EventId is ignored when Cluster is specified")
            self.ClusterId = Cluster.id
            return
        if Event:
            if ClusterId or EventId:
```

Reply:
EventId could be 0, so we would not this way, thanks for suggestion.